crates/bevy_app/src/app_builder.rs

```diff
@@ -273,13 +427,44 @@ impl AppBuilder {
         self
     }
 
+    /// Adds a group of plugins
+    ///
+    /// Bevy plugins can be grouped into a set of plugins. By default
+    /// bevy provides a few lists of plugins that can be used to kickstart
+    /// the development.
+    ///
+    /// Current plugins offered are [`DefaultPlugins`] and [`MinimalPlugins`].
+    ///
+    /// ## Example
+    /// ```
+    /// # use bevy_app::prelude::*;
+    /// #
+    /// App::build();
+    ///     //.add_plugins(MinimalPlugins)
+    /// ```
     pub fn add_plugins<T: PluginGroup>(&mut self, mut group: T) -> &mut Self {
         let mut plugin_group_builder = PluginGroupBuilder::default();
         group.build(&mut plugin_group_builder);
         plugin_group_builder.finish(self);
         self
     }
 
+    /// Adds a group of plugins with an initializer method
+    ///
+    /// Can be used to add a group of plugins, where the group is modified
+    /// before insertion into Bevy application. For example, you can add
+    /// extra plugins at a specific place in the plugin group, or deactivate
+    /// specific plugins while keeping the rest.
+    ///
+    /// ## Example
+    /// ```
+    /// # use bevy_app::prelude::*;
+    /// #
+    /// App::build();
+    ///     // .add_plugins_with(DefaultPlugins, |group| {
+    ///            // group.add_before::<bevy::asset::AssetPlugin, _>(MyOwnPlugin)
+    ///        // })
```

Comment:
```suggestion
    /// # use bevy_app::{prelude::*, PluginGroupBuilder};
    /// #
    /// # // Dummies created to avoid using bevy_internal which pulls in to many dependencies.
    /// # struct DefaultPlugins;
    /// # impl PluginGroup for DefaultPlugins {
    /// #     fn build(&mut self, group: &mut PluginGroupBuilder){
    /// #         group.add(bevy_log::LogPlugin::default());
    /// #     }
    /// # }
    /// #
    /// # struct MyOwnPlugin;
    /// # impl Plugin for MyOwnPlugin {
    /// #     fn build(&self, app: &mut AppBuilder){;}
    /// # }
    /// #
    /// App::build()
    ///      .add_plugins_with(DefaultPlugins, |group| {
    ///             group.add_before::<bevy_log::LogPlugin, _>(MyOwnPlugin)
    ///         });
```